app_class.rb

```diff
@@ -0,0 +1,129 @@
+require 'json'
+require 'date'
+require_relative 'item_class'
+require_relative 'book_class'
+require_relative 'album_class'
+require_relative 'game_class'
+require_relative 'label_class'
+require_relative 'genre_class'
+require_relative 'author_class'
+require_relative 'storage_module'
+
+class App
+  include Storage
+
+  def initialize
+    @classes = [Book, MusicAlbum, Game, Label, Genre, Author]
+    @lists = { 'book' => [], 'album' => [], 'game' => [], 'label' => [], 'genre' => [], 'author' => [] }
+    @options = %w[book album game label genre author]
+    load_lists
+  end
+
+  def load_lists
+    @lists.reverse_each do |key, value|
+      items = read_file(key)
+      items.each do |item|
+        object = list_arguments(key, item)
+        create_relations(object, item) if %w[book album game].include?(key)
+        value.append(object)
+      end
+    end
+  end
+
+  def list_arguments(type, arguments)
+    case type
+    when 'book'
+      Book.new(arguments['@publisher'], arguments['@cover_state'], arguments['@publish_date'], arguments['@id'])
+    when 'album'
+      MusicAlbum.new(arguments['@on_spotify'], arguments['@publish_date'], arguments['@id'])
+    when 'game'
+      Game.new(arguments['@multiplayer'], arguments['@last_played_at'], arguments['@publish_date'], arguments['@id'])
+    when 'label'
+      Label.new(arguments['title'], arguments['color'], arguments['id'])
+    when 'genre'
+      Genre.new(arguments['name'], arguments['id'])
+    when 'author'
+      Author.new(arguments['first_name'], arguments['last_name'], arguments['id'])
+    end
+  end
```

Comment:
- [ ] You all have done fantastic work, just a few thing to go, when I enter a bad date, the app crashes, can you make it show an error message and not crash, because it is not good user experience
![image](https://github.com/mahammad-mostafa/things-catalogue/assets/66097371/925db7a3-a5cf-496e-b1fb-a53f037f772a)